### PR TITLE
Proper compilation of Metas (GHC and JS)

### LINKED ISSUE #5754: Internal error when compiling program with quoted metavariable
(forked off from the discussion at https://github.com/agda/agda/issues/5731)
```agda
open import Agda.Builtin.IO
open import Agda.Builtin.List
open import Agda.Builtin.Reflection renaming (bindTC to _>>=_)
open import Agda.Builtin.String
open import Agda.Builtin.Unit

macro
  getMeta : Term → TC ⊤
  getMeta hole@(meta m _) = quoteTC m >>= unify hole
  getMeta _ = typeError []

myMeta : Meta
myMeta = getMeta

postulate putStrLn : String → IO ⊤
{-# FOREIGN GHC import qualified Data.Text as T #-}

main : IO ⊤
main = putStrLn (primShowMeta myMeta)
```
Trying to compile this with the GHC backend this produces an internal error:
```
An internal error has occurred. Please report this as a bug.
Location of the error: __IMPOSSIBLE__, called at src/full/Agda/Compiler/MAlonzo/Compiler.hs:1097:18 in Agda-2.6.3-2GgStRv6exQAe2hnultIdP:Agda.Compiler.MAlonzo.Compiler
```

P.S. (Andreas): Error is thrown here:
https://github.com/agda/agda/blob/5f074ffbb86024f32f94f7b2372d2cfaeda73d9e/src/full/Agda/Compiler/MAlonzo/Compiler.hs#L1110

### PR DESCRIPTION
**Pull request by fredrikNordvallForsberg**
Fixes #5754. Investigated together with @felixwellen at AIM XXXI.

### CODE REVIEW

File: src/data/JS/agda-rts.js [hunk 1]
```diff
@@ -374,3 +374,18 @@ exports.primShowQName = x => x['name'];
...
 // primQNameFixity : Name -> Fixity
 exports.primQNameFixity = x => x['fixity'];
+
+// Meta
+
+// primShowMeta : Meta -> String
+exports.primShowMeta = x => "_" + x['id'] + "@" + x['module'];
+
+// primMetaToNat : Meta -> Nat
+exports.primMetaToNat = x => x['module'] * 2^64 + x['id'];
```
Comment (andreasabel) [inline src/data/JS/agda-rts.js:387]: Should this calculation be kept in sync with similar code in other backends or the reflection code?  If yes, please add a comment pointing to the relevant locations?

File: src/full/Agda/Compiler/MAlonzo/Primitives.hs [hunk 2]
```diff
@@ -286,10 +286,10 @@ primBody s = maybe unimplemented (fromRight (hsVarUQ . HS.Ident) <$>) $
...
   , "primQNameToWord64s"  |-> return "\\ qn -> (MAlonzo.RTE.nameId qn, MAlonzo.RTE.moduleId qn)"
   , "primQNameToWord64sInjective" |-> return mazErasedName
-  , "primMetaEquality"    |-> rel "(==)" "Integer"
-  , "primMetaLess"        |-> rel "(<)" "Integer"
-  , "primShowMeta"        |-> return "\\ x -> Data.Text.pack (\"_\" ++ show (x :: Integer))"
-  , "primMetaToNat"       |-> return "(id :: Integer -> Integer)"
+  , "primMetaEquality"    |-> rel "(==)" "(Integer, Integer)"
+  , "primMetaLess"        |-> rel "(<)" "(Integer, Integer)"
+  , "primShowMeta"        |-> return "\\ (m, h) -> Data.Text.pack (\"_\" ++ show (m :: Integer) ++ \"@\" ++ show (h :: Integer))"
+  , "primMetaToNat"       |-> return "\\ (m, h) -> (h :: Integer) * 2^64 + (m :: Integer)"
```
Comment (andreasabel) [inline src/full/Agda/Compiler/MAlonzo/Primitives.hs:295]: See respective comment for JS backend.

**Comment by andreasabel** [comment]
Considered for 2.6.3.

File: src/data/JS/agda-rts.js (hunk 1 above)
Comment (fredrikNordvallForsberg) [reply src/data/JS/agda-rts.js:387]: Good point, added.

File: src/full/Agda/Compiler/MAlonzo/Primitives.hs (hunk 2 above)
Comment (fredrikNordvallForsberg) [reply src/full/Agda/Compiler/MAlonzo/Primitives.hs:295]: Also done.

**Review by andreasabel — APPROVED** [review]
Great, thanks!

**Comment by andreasabel** [comment]
Cherry-picked to `release-2.6.3`.

**Pull request merged**